alibi/explainers/similarity/grad.py

```diff
@@ -249,7 +250,8 @@ def explain(
         X, Y = self._preprocess_args(X, Y)
```

Comment:
It may be easier to have a separate `_preprocess_args` function dependent on data type and dispatch to that rather than having lots of conditional statements throughout.